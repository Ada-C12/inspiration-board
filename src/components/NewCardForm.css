/* The Modal (background) */

.new-card-form {
  margin: 1rem;
  padding-bottom: 4rem;
}

.new-card-form__header {
  text-align: center;
  text-transform: uppercase;
}

.new-card-form__form {
  font-size: 1.5em;
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-auto-flow: row;
}

.new-card-form__form-label  {
  grid-column: labels;
  grid-row: auto;
}

.new-card-form__form-textarea {
  grid-column-start: span 2;
}
.new-card-form__form-select,
.new-card-form__form-textarea,
.new-card-form__form-button {
  font-family: 'Raleway', sans-serif;
  font-size: 1.25em;
  grid-row: auto;
}

.new-card-form__form-button {
  background-color: inherit;
  border: 1px solid black;
  font-size: 1em;
}
