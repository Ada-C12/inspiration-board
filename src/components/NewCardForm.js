import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';
import './NewCardForm.css';

const emojis = require("emoji-dictionary");
const EMOJI_LIST = ["", "heart_eyes", "beer", "clap", "sparkling_heart", "heart_eyes_cat", "dog"]

class NewCardForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      text: "", 
      emoji: ""
    };
  }

  onChange = (event) => {
    const { name, value } = event.target;
    
    const updatedState = {};
    updatedState[name] = value;

    this.setState(updatedState);
  };

  onSelect = event => {
    this.setState({
      emoji: event.target.value
    });
  };
  
  onSubmit = (e) => {
    e.preventDefault();

    let newSubmission = {}
    Object.keys(this.state).forEach((key) => {
      if ( key === 'Emoji' || key === "emoji" ) {
        const value = this.state[key]    
        newSubmission[key] = emoji.getName(value)
      } else {
        newSubmission[key] = this.state[key]
      }
    }); 

    this.props.addCard(newSubmission);

    this.setState({ 
      text: "", 
      emoji: "" 
    });
  }

  render() {
    const emojiList = EMOJI_LIST.map((emoji) => {
      return <option> {emojis.getUnicode(emoji)}</option>
    });

    return (
        <form className="new-card-form">
          <h1 className="new-card-form__header"> Create a new card! </h1>
          <div className="new-card-form__form">
            <input 
              className="new-card-form__form-textarea"
              type="text" 
              name="text" 
              placeholder="Text" 
              value={this.state.text}
              onChange={this.onChange}
            />
            < select
              className="new-card-form__form-select"
              value={this.state.emoji}
              onChange={this.onSelect}>
              {emojiList}
            </select>
            <input 
              className="new-card-form__form-button"
              type="submit" 
              value="Submit" 
              onClick={this.onSubmit} 
            />
          </div>
        </form> 
    )
  }
}

export default NewCardForm;
