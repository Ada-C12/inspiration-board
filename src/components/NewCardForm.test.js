import React from 'react';
import { render, cleanup } from '@testing-library/react'
import NewCardForm from './NewCardForm';

describe ('NewCardForm', () => {
  test('it matches existing snapshot', () => {
    /* Arrange - Act */
    const { asFragment } = render(
      <NewCardForm 
      addCardCallback={() => { }}
      />
    );

    /* Assert */
    expect(asFragment()).toMatchSnapshot();
    cleanup();
  })
})