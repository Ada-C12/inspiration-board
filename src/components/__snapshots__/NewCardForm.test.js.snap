// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewCardForm that it matches the existing snapshot 1`] = `
<DocumentFragment>
  <div
    class="NewCardForm"
  >
    <h3
      class="new-card-form__header"
    >
      Submit Inspiration Note
    </h3>
    <form
      class="new-card-form__form"
    >
      <div
        class="new-card-form__form-label"
      >
        <input
          class="new-card-form__form-textarea"
          name="text"
          placeholder="Inspirational note here"
          type="text"
          value=""
        />
      </div>
      <div
        class="new-card-form__form-label"
      >
        <input
          class="new-card-form__form-select"
          name="emoji"
          value=""
        />
      </div>
      <div
        class="NewCardForm__submit"
      >
        <input
          class="new-card-form__form-button"
          type="submit"
          value="Submit Note"
        />
      </div>
    </form>
  </div>
</DocumentFragment>
`;
